### PR TITLE
Monkey re-sprite & Bonus procedural monkey clothing

### PR DESCRIPTION
**Pull request by DrDuckedGoose**
## About The Pull Request

Updates monkey sprites to a semi-modern standard. 

Gives monkeys the option to wear jumpsuits, or any type of ``i_clothing``. 
This is done by clipping down the human worn-sprite and making some slight adjustments through shifting & stretching. This is also applied to hats, bags & masks. 

Monkey-icons are only generated if a monkey tries to equip a piece of clothing. Once an icon has been generated, all similar items will use that icon, instead of making their own, when a monkey equips them.

Additionally introduces a new gimmick role, Experiment. This role will have the player spawn as a monkey with a science outfit & ID.

## Why It's Good For The Game

Improves crusty sprites & emphasizes monkey RP without introducing any force-rp mechanics.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>
Old

image

New

image
image
image
image
image
image
image
image
image
image
image
image

</details>

## Changelog
:cl:
add: New monkey sprites
tweak: Tweaks monkey inventory update - procs interface with ``monkey_icon``
add: Add ``i_clothing`` slot to monkey equipment slots
tweak: Allow monkeys to equip ``i_clothing``
add: Add Experiment gimmick job
add: Add Experiment job-spawn to MetaStation
/:cl:

### CODE REVIEW

**Comment by DrDuckedGoose** [comment]
My git-head must be fucked, pulling commits from old PRs

**Comment by mcmeiler** [comment]
This looks more like goon monkeys than anything. Pass from me.

**Comment by Isy232** [comment]
Great, now monkeys dont look so obviously 2 dimensional

**Comment by dwasint** [comment]
I like it sans the sideways sprite, but than again making them true 3/4th would not be in scope for this pr

**Comment by FortSumter** [comment]
Monkey station

**Comment by DerpiestBruh** [comment]
YES DUDE YESSSSSSSSSSSSS

**Comment by Fronsis** [comment]
Monke with da drip will become real in FIVE SECONDS

Though i'd keep the original color eyes

**Comment by ophaq** [comment]
If this got merged into TG I'd probably make monkey station at some point for funny.

File: code/_onclick/hud/monkey.dm [hunk 1]
```diff
@@ -44,6 +44,16 @@
 	using.hud = src
 	static_inventory += using
 
+	inv_box = new /atom/movable/screen/inventory()
+	inv_box.name = "uniform"
+	inv_box.icon = ui_style
+	inv_box.icon_state = "uniform"
+//	inv_box.icon_full = "template"
```
Comment (itsmeow) [inline code/_onclick/hud/monkey.dm]: commented code

File: code/modules/jobs/job_types/gimmick.dm [hunk 2]
```diff
@@ -165,3 +165,42 @@
 	uniform = /obj/item/clothing/under/suit/black_really
 	shoes = /obj/item/clothing/shoes/laceup
 	can_be_admin_equipped = TRUE
+
+/datum/job/gimmick/experiment
+	title = "Experiment"
```
Comment (itsmeow) [inline code/modules/jobs/job_types/gimmick.dm]: JOB_NAME_EXPERIMENT

File: code/modules/jobs/job_types/gimmick.dm [hunk 3]
```diff
@@ -165,3 +165,42 @@
...
+	flag = EXPERIMENT
+	supervisors = "the research director"
+	department_head = list("Research Director")
```
Comment (itsmeow) [inline code/modules/jobs/job_types/gimmick.dm]: `JOB_NAME_RESEARCH_DIRECTOR`

File: code/modules/mob/living/carbon/update_icons.dm [hunk 4]
```diff
@@ -160,7 +160,7 @@
 	if(head)
 		overlays_standing[HEAD_LAYER] = head.build_worn_icon(default_layer = HEAD_LAYER, default_icon_file = 'icons/mob/clothing/head.dmi')
 		update_hud_head(head)
-
+
```
Comment (itsmeow) [inline code/modules/mob/living/carbon/update_icons.dm:163]: useless whitespace addition

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by itsmeow — DISMISSED** [review]

**Comment by itsmeow** [comment]
check machine broke

```
code/__DEFINES/jobs.dm, line 200, column 9:
 warning: macro redefined: JOB_NAME_EXPERIMENT
- 134:9: previous definition of JOB_NAME_EXPERIMENT
- ```

**Review by itsmeow — DISMISSED** [review]

**Comment by itsmeow** [comment]
bleh
```
file ./code/__DEFINES/misc.dm is missing a trailing newline
```

**Comment by dwasint** [comment]
Few things,
1. I think de hard coding the path would be worthwhile as gag items are usually in their own files for easy of editing
2. This probably doesn't support gag items period, and having yet another reason to not have gags is kinda eh

**Comment by DrDuckedGoose** [comment]
> Few things,
> 
> 1. I think de hard coding the path would be worthwhile as gag items are usually in their own files for easy of editing
> 2. This probably doesn't support gag items period, and having yet another reason to not have gags is kinda eh

It's almost like GAGs is a shitty system to work with 🤔

**Comment by DrDuckedGoose** [comment]
Wait, it already, kinda, does?
image
**MY WORK HERE IS DONE!**

**Comment by dwasint** [comment]
> > Few things,
> > 
> > 1. I think de hard coding the path would be worthwhile as gag items are usually in their own files for easy of editing
> > 2. This probably doesn't support gag items period, and having yet another reason to not have gags is kinda eh
> 
> It's almost like GAGs is a shitty system to work with 🤔

Gags is a great system to work with once you play around with it, being able to recolor certain parts of items instead of having like 30 different color combos bloating sprites is great

**Review by PowerfulBacon — APPROVED** [review]

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Actually, there is no way there should be a roundstart monkey gimmick job. Having a 25% chance to have an unplayable round apart from being forced to go to genetics is dumb, especially since you can't choose

**Review by PowerfulBacon — APPROVED** [review]

**Comment by itsmeow** [comment]
Runtime
image

**Pull request merged**